### ISSUE TITLE
Add a test... 'tis so bad that it doesn't have a test

### ISSUE THREAD

**Issue by dance2die** [dance2die/react-use-localstorage#19]
If anyone's willing to help out, go for it 🙂

**Comment by SeanMcP** [comment]
I took a look this evening. It was pretty straightforward to get "100% coverage" (as far as Jest was concerned), but I ran into some issues mocking `localStorage`. I'll keep at it.

You can view the changes here: https://github.com/SeanMcP/react-use-localstorage

**Comment by dance2die** [comment]
Thank you, @SeanMcP 😃

I just tried your fork and looks great~
Would you be able to do a PR?

**Comment by SeanMcP** [comment]
Happy to help!

Right now it just tests that `useState()` works predictably, which I don't feel adds much value. Since the library is specifically for saving data in `localStorage`, I would want the tests to include that before making a pull request.

What do you think?

**Comment by dance2die** [comment]
You are right. The test file tests the React library (`useState`), not what `react-use-localstorage` does 🤔

>  I would want the tests to include that before making a pull request.

Yes, would you include the test for localStorage before a PR?
_Or you can do a [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/), which creates fake PR, not ready for review, yet 😀_

**Comment by SeanMcP** [comment]
I tried using [`jest-localstorage-mock`](https://www.npmjs.com/package/jest-localstorage-mock), but it broke all of the tests. That leads me to believe it has something to do with the setup, so I'm planning to take another look later today.

I will commit the in-progress changes to my fork so that you can keep an eye on it too.

**Comment by SeanMcP** [comment]
I ended up removing `jest-localstorage-mock` and creating my own. It may have to be revisited if/when API changes utilize `localStorage` more.

**Comment by dance2die** [comment]
That `LocalStorageMock` class looks like a great alternative to using a library, `jest-localstorage-mock` as it's small enough and readable
